### ISSUE TITLE
Preview error when using an Umb.PropertyEditorUi.ContentPicker on 2.0.2 / 14.3.0

### ISSUE THREAD

**Issue by alangilby** [rickbutterfield/Our.Umbraco.BlockPreview#74]
### Which Umbraco.Community.BlockPreview version are you using?

2.0.2

### Which Umbraco version are you using? For example: 10.4.0 - don't just write v10

14.3.0

### Bug summary

When previewing an Umb.PropertyEditorUi.ContentPicker property the following errors is returned.

`System.FormatException: String "[
  {
    "type": "document"" is not a valid udi.
   at Umbraco.Cms.Core.UdiParser.ParseInternal(String s, Boolean tryParse, Boolean knownTypes, Udi& udi)
   at Umbraco.Cms.Core.UdiParser.Parse(String s)
   at System.Linq.Enumerable.SelectArrayIterator'2.Fill(ReadOnlySpan'1 source, Span'1 destination, Func'2 func)
   at System.Linq.Enumerable.SelectArrayIterator'2.ToArray()
   at Umbraco.Cms.Core.PropertyEditors.ValueConverters.MultiNodeTreePickerValueConverter.ConvertSourceToIntermediate(IPublishedElement owner, IPublishedPropertyType propertyType, Object source, Boolean preview)
   at Umbraco.Cms.Core.Models.PublishedContent.PublishedPropertyType.ConvertSourceToInter(IPublishedElement owner, Object source, Boolean preview)
   at Umbraco.Cms.Core.PublishedCache.PublishedElementPropertyBase.GetInterValue()
   at Umbraco.Cms.Core.PublishedCache.PublishedElementPropertyBase.GetValue(String culture, String segment)
   at Umbraco.Extensions.PublishedPropertyExtension.Value[T](IPublishedProperty property, IPublishedValueFallback publishedValueFallback, String culture, String segment, Fallback fallback, T defaultValue)
   at Umbraco.Extensions.PublishedElementExtensions.Value[T](IPublishedElement content, IPublishedValueFallback publishedValueFallback, String alias, String culture, String segment, Fallback fallback, T defaultValue)
   at Umbraco.Extensions.FriendlyPublishedElementExtensions.Value[T](IPublishedElement content, String alias, String culture, String segment, Fallback fallback, T defaultValue)
   at AspNetCore.Views_Partials_blockgrid_Components_blockFeatures.ExecuteAsync() in ---\Views\Partials\blockgrid\Components\blockFeatures.cshtml:line 12
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Umbraco.Community.BlockPreview.Services.BlockPreviewService.GetMarkupFromPartial(ControllerContext controllerContext, ViewDataDictionary viewData, String contentAlias, BlockType blockType)
   at Umbraco.Community.BlockPreview.Services.BlockPreviewService.GetMarkupFromPartial(ControllerContext controllerContext, ViewDataDictionary viewData, String contentAlias, BlockType blockType)
   at Umbraco.Community.BlockPreview.Services.BlockPreviewService.GetMarkup(ControllerContext controllerContext, String contentAlias, ViewDataDictionary viewData, BlockType blockType)
   at Umbraco.Community.BlockPreview.Services.BlockPreviewService.RenderGridBlock(String blockData, ControllerContext controllerContext, String blockEditorAlias, Guid documentTypeUnique, String contentUdi, String settingsUdi)
   at Umbraco.Community.BlockPreview.Controllers.BlockPreviewApiController.PreviewGridBlock(String blockData, Guid nodeKey, String blockEditorAlias, String contentElementAlias, String culture, Guid documentTypeUnique, String contentUdi, String settingsUdi)'
`
As soon as content is picked the preview throws the above error, the partial view displays okay when not previewing.

### Steps to reproduce

![error-screen1](https://github.com/user-attachments/assets/1b5ae9b4-8fe1-42d2-8d42-32a3f8519407)
![without-picker-content](https://github.com/user-attachments/assets/1055b5bd-a68e-4e1f-8a4b-3605a36bebf7)
Partial view code as follows:
```@using Ext.Helpers
@using Umbraco.Community.BlockPreview.Extensions

@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage<Umbraco.Cms.Core.Models.Blocks.BlockGridItem<BlockFeatures>>

@if (Context.Request.IsBlockPreviewRequest())
{
    if (Model.Content is BlockFeatures typedModel)
    {
        <p>@typedModel.Title</p>

        var manuallyPicked = typedModel.Value<IEnumerable<IPublishedContent>>("manuallyPicked");

    }


}
else
{
    if (Model.Content is BlockFeatures typedModel)
    {
        if (typedModel.ManuallyPicked != null)
        {

            <div class="grid grid-cols-12 gap-4">
                @foreach (var contentItem in typedModel.ManuallyPicked)
                {

                    <div class="col-span-6 md:col-span-4 lg:col-span-3">
                        @await Html.PartialAsync("/views/partials/contentPage/overlaid-card.cshtml", contentItem as ContentPage)
                    </div>
                }

            </div>
        }
    }
}


### Expected result / actual result

Preview as per front end templates

**Comment by kevin9092** [comment]
I tried to replicate this, but I couldn't get this issue. I must be missing something.

Could you set up a mini-repository or add additional code snippets? I think that would make it much easier to debug the problem you are experiencing.

**Comment by alangilby** [comment]
Thanks for taking a look, There's reference to the same error in this discord https://discord-chats.umbraco.com/t/22779483/umbraco-14-block-grid-editor-customisation (search for "document" is not a valid UDI on that page)

I wonder if it's a related issue (and a problem with the umbraco picker)

I'm not sure what other code snippets i can provide as it's 'fairly basic' implementation. 

This is the code from \Views\Partials\blockgrid\Components\blockGridFeatures.cshtml stripped down to as basic as i can, just doing a null check is enough to throw the above error for me.

`@using Umbraco.Community.BlockPreview.Extensions
@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage<Umbraco.Cms.Core.Models.Blocks.BlockGridItem>
@if (Model.Content is BlockGridFeatures typedModel)
{
    @if (typedModel.ManuallyPicked != null)
    {
        <p>Not null</p>
    }
    <h3>@typedModel.Title</h3>
}
`
the block element and picker config as per screens:

![Umbraco-10-16-2024_12_45_PM](https://github.com/user-attachments/assets/bb22198e-5da9-436c-89da-98b36a73b7c3)

![content-picker](https://github.com/user-attachments/assets/9d5d22df-7ad8-4b42-a7c4-bb6b311094cb)

**Comment by kevin9092** [comment]
@alangilby Thanks for all the information.

I have managed to replicate this and have submitted a pull request (#76) that should resolve this issue. Umbraco is expecting a UDI e.g. `umb://element/<guid>` which was currently only passing through `<guid>`.

Due to the nature of the fix, I have requested the repo owner have a look through it just in case I am missing potential edge-cases.

**Comment by rickbutterfield** [comment]
@alangilby The fix for this is included in 2.0.3 which is currently being released to NuGet. Please give it a try and report back!

**Comment by alangilby** [comment]
This looks to have worked for me Rick, many thanks. 

I'm also seeing the same issue in 14.3.1 as per https://github.com/rickbutterfield/Umbraco.Community.BlockPreview/issues/80 but i don't think it's related to this fix as i rolled back to 2.0.2 and same issue, so think it's 14.3.1 related. 

Thanks for the quick fix of this one.
